docs/book/src/images/worker-machines-resources.plantuml

```diff
@@ -0,0 +1,48 @@
+@startuml worker-machines-resources
+
+skinparam nodesep 80
+skinparam ranksep 80
+skinparam object {
+    backgroundColor<< Infrastructure provider CRD >> #D9EAD3
+    backgroundColor<< Bootstrap provider CRD >> #FEF4F4
+}
+
+' Cluster API CRD
+object Cluster << Cluster API CRD >>
+object MachineDeployment << Cluster API CRD >> {
+    spec.clusterName
+    <font color="Green">spec.template.spec.infrastructureRef</font>
+    <font color="Red">spec.template.spec.bootstrap.configRef</font>
+}
+object MachineSet << Cluster API CRD >> {
+    <font color="Green">spec.template.spec.infrastructureRef</font>
+    <font color="Red">spec.template.spec.bootstrap.configRef</font>
+}
+object Machine << Cluster API CRD >> {
+    <font color="Green">spec.infrastructureRef</font>
+    <font color="Red">spec.bootstrap.configRef</font>
+}
+
+' Bootstrap provider CRD
+object BarBootstrapConfig << Bootstrap provider CRD >> 
+object BarBootstrapConfigTemplate << Bootstrap provider CRD >>
+
+' Infrastructure provider CRD
+object FooMachineTemplate << Infrastructure provider CRD >>
+object FooMachine << Infrastructure provider CRD >>
```

Comment:
It would be great if we can keep FooMachineTemplate and FooMachine on the same side of the graph..

Reply:
Hmmm, I'm not sure how to do this in PlantUML.

Reply:
You can't with PlantUML, it's all auto-generated